src/main/java/com/ecommerce/ECommerceApp/model/PhoneNumber.java

```diff
@@ -19,7 +21,7 @@ public void setPhoneNumber(String phoneNumber) {
         if (isValid(phoneNumber)) {
             this.phoneNumber = phoneNumber;
         } else {
-            throw new IllegalArgumentException("Invalid phone number: " + phoneNumber);
+            throw new InvalidPhoneNumberException("Invalid phone number " + phoneNumber);
```

Comment:
Implemented custom exception handling for invalid phone number to help make code more readable and clearer on what is being handled during an exception being thrown.

Source: https://www.baeldung.com/java-clean-code